scope.go

```diff
@@ -0,0 +1,76 @@
+package flux
```

Comment:
Why is this in the `flux` package? Is it used outside the `interpreter` package at all?

Reply:
The interpreter only uses the Scope interface. The only place where we create a concrete implementation of the Scope interface is for defining the prelude in the `flux` package so I decided to place it there.